src/bio/bio_buffer.c

```diff
@@ -637,16 +637,26 @@ rw_completion(void *cb_arg, int err)
 	D_ASSERT(xs_ctxt->bxc_blob_rw > 0);
 	xs_ctxt->bxc_blob_rw--;
 
-	if (biod->bd_result == 0 && err != 0) {
-		biod->bd_result = daos_errno2der(-err);
+	if (biod->bd_update)
```

Comment:
Test is failing with this code change.  rw_completion() is not getting called for each insert_single_with_rxnr() or lookup_single_with_rxnr() not sure why. writeErr:true is getting set during pool creation too without setting the error.

Reply:
Changing the code to defect BIO error with the initial reporting if loop.